src/storage/btree.c

```diff
@@ -4280,13 +4302,171 @@ btree_read_record (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pgptr, REC
+    {
+      return NO_ERROR;
+    }
+
+  assert (pgptr == bts->C_page && VPID_EQ (pgbuf_get_vpid_ptr (pgptr), &bts->C_vpid));
+
+  if (bts->common_prefix_size >= 0)
+    {
+      assert (VPID_EQ (&(bts->C_vpid), &bts->cur_common_prefix_page_vpid));
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+      assert (LSA_EQ (&bts->cur_common_prefix_page_lsa, pgbuf_get_lsa (pgptr)));
+      assert (bts->common_prefix_size == btree_node_get_common_prefix (thread_p, &bts->btid_int, pgptr));
+#endif
+
+      if (bts->common_prefix_size > 0)
+	{
+	  bts->is_cur_key_compressed = true;
+	}
+      return NO_ERROR;
+    }
+
+  assert (bts->common_prefix_size == COMMON_PREFIX_UNKNOWN);
+
+  /* recalculate n_prefix */
+  n_prefix = btree_node_get_common_prefix (thread_p, &bts->btid_int, pgptr);
+
+  bts->common_prefix_size = n_prefix;
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+  VPID_COPY (&(bts->cur_common_prefix_page_vpid), &(bts->C_vpid));
+  LSA_COPY (&bts->cur_common_prefix_page_lsa, pgbuf_get_lsa (pgptr));
+#endif
+  btree_clear_key_value (&(bts->clear_common_prefix_key), &(bts->common_prefix_key));
+
+  if (n_prefix > 0)
+    {
+      RECDES peek_rec;
+      LEAF_REC leaf_pnt;
+      int dummy_offset;
+
+      (void) spage_get_record (thread_p, pgptr, 1, &peek_rec, PEEK);
+      error =
+	btree_read_record_without_decompression (thread_p, &bts->btid_int, &peek_rec, &(bts->common_prefix_key),
+						 &leaf_pnt, BTREE_LEAF_NODE, &(bts->clear_common_prefix_key),
+						 &dummy_offset, PEEK_KEY_VALUE);
+      if (error != NO_ERROR)
+	{
+	  btree_clear_key_value (&(bts->clear_common_prefix_key), &(bts->common_prefix_key));
+	  return error;
+	}
+      assert (btree_leaf_is_flaged (&peek_rec, BTREE_LEAF_RECORD_FENCE));
+      assert (DB_VALUE_TYPE (&(bts->common_prefix_key)) == DB_TYPE_MIDXKEY);
+      bts->is_cur_key_compressed = true;
+    }
+  else
+    {
+      if (n_prefix < 0)
+	{
+	  assert_release (false);
 	  return n_prefix;
 	}
     }
 
   return NO_ERROR;
 }
 
+static void
+btree_make_complete_key_including_prefix (BTREE_SCAN * bts, DB_VALUE * common_prefix_key, int common_prefix_size)
+{
+  int error = NO_ERROR;
+  if (common_prefix_size > 0)
+    {
+      DB_VALUE completed_key;
+      db_make_null (&completed_key);
+
+      assert (bts->is_cur_key_compressed == true);
+      error = pr_midxkey_add_prefix (&completed_key, common_prefix_key, &(bts->cur_key), common_prefix_size);
+      if (error != NO_ERROR)
+	{
+	  assert_release (false);
+	  return;
+	}
+
+      btree_clear_key_value (&(bts->clear_cur_key), &(bts->cur_key));
+      bts->cur_key = completed_key;
+      bts->clear_cur_key = true;
+      bts->is_cur_key_compressed = false;
+    }
+}
+
+/*
+ * btree_check_decompress_key () - 
+ *               This is a function to convert the two parts, cur_key and common_prefix_key, into a complete key by combining them with cur_key.
+ *   return:
+ *   bts(in):
+ */
+static void
+btree_check_decompress_key (BTREE_SCAN * bts)
```

Comment:
What is the purpose of this function?

Reply:
Check it and make it if necessary.
The name is not appropriate.

How about btree_make_after_checking_completed_key()?

Reply:
I think the functions for check and make purposes should be separate.

Reply:
The purpose is to express my opinion, so if you read it, please resolve it.

Reply:
I'll keep it as is for now and review it one more time in [CBRD-25159]
Thank you.